client/lib/titan/get-titan-urls.js

```diff
@@ -4,20 +4,21 @@
 const TITAN_APPS = {
 	CALENDAR: 'calendar',
 	CONTACTS: 'contacts',
-	EMAIL: 'email',
+	EMAIL: 'mail',
 };
 
 /**
  * Generates a URL pointing to the given Titan App
  *
+ * @param {undefined|{titanMailSubscription?:{appsUrl?: string}}} domain - Domain object
  * @param {string?} email - The email address of the Titan account. Used for autofill on Titan's login page.
  * @param {string?} app - Can be one of the `TITAN_APPS` - `email`, `calendar` or `contacts`
- * @param {boolean?} clearPreviousSessions - Whether to clear previously logged in sessions.
+ * @param {boolean?} clearPreviousSessions - Whether to clear previously logged-in sessions.
  * @returns The URL with app and prefilled `email_account` as query parameter
  */
-function getTitanUrl( email, app = TITAN_APPS.EMAIL, clearPreviousSessions = false ) {
-	const titanBaseUrl = 'https://wp.titan.email';
-	const titanAppUrl = new URL( `${ titanBaseUrl }/${ app }` );
+function getTitanUrl( domain, email, app = TITAN_APPS.EMAIL, clearPreviousSessions = false ) {
```

Comment:
Personal preference: after seeing a bunch of code all passing a `domain` object around, where that object may or may not have anything to do with the current email, I am not really liking the use of `domain` directly. (This grew on me as I was reviewing the code.)

My preference is to aim for something like the following (but this is not a blocker):
```
function getTitanUrl( email, appsUrlPrefix = 'https://wp.titan.email', app = TITAN_APPS.EMAIL, clearPreviousSessions = false ) {
	const titanBaseUrl = appsUrlPrefix ?? 'https://wp.titan.email';
	const titanAppUrl = new URL( `${ titanBaseUrl }/${ app }/` );
...

function getTitanAppsUrlPrefix( domain: ResponseDomain | null | undefined ): string {
	return domain?.titanMailSubscription?.appsUrl ?? 'https://wp.titan.email'; // this string may make sense as a constant
}

// then callers do something like the following:
const url = getTitanEmailUrl( email, getTitanAppsUrlPrefix( firstTitanDomain ), true );
```
My proposed approach does push the prefix computation onto callers, but it also means that we're passing the actual data we need down the stack, rather than an object that isn't actually related to the email we're building a link for.